distribution/services/update.go

```diff
@@ -0,0 +1,145 @@
+package services
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	rthttpclient "github.com/jfrog/jfrog-client-go/artifactory/httpclient"
+	artifactoryUtils "github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/auth"
+	distrbutionServiceUtils "github.com/jfrog/jfrog-client-go/distribution/services/utils"
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+)
+
+type ReleaseNotesSyntax string
+
+const (
+	Markdown  ReleaseNotesSyntax = "markdown"
+	Asciidoc                     = "asciidoc"
+	PlainText                    = "plain_text"
+)
+
+type UpdateReleaseBundleService struct {
+	client      *rthttpclient.ArtifactoryHttpClient
+	DistDetails auth.CommonDetails
+	DryRun      bool
+}
+
+func NewUpdateReleaseBundleService(client *rthttpclient.ArtifactoryHttpClient) *UpdateReleaseBundleService {
+	return &UpdateReleaseBundleService{client: client}
+}
+
+func (ur *UpdateReleaseBundleService) GetDistDetails() auth.CommonDetails {
+	return ur.DistDetails
+}
+
+func (ur *UpdateReleaseBundleService) UpdateReleaseBundle(createBundleParams CreateUpdateReleaseBundleParams) error {
+	releaseBundleBody, err := CreateBundleBody(createBundleParams, ur.DryRun)
+	if err != nil {
+		return err
+	}
+	return ur.execUpdateReleaseBundle(createBundleParams.Name, createBundleParams.Version, createBundleParams.GpgPassphrase, releaseBundleBody)
+}
+
+func (ur *UpdateReleaseBundleService) execUpdateReleaseBundle(name, version, gpgPassphrase string, releaseBundle *ReleaseBundleBody) error {
+	httpClientsDetails := ur.DistDetails.CreateHttpClientDetails()
+	content, err := json.Marshal(releaseBundle)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+	url := ur.DistDetails.GetUrl() + "api/v1/release_bundle/" + name + "/" + version
+	distrbutionServiceUtils.SetGpgPassphrase(gpgPassphrase, &httpClientsDetails.Headers)
+	artifactoryUtils.SetContentType("application/json", &httpClientsDetails.Headers)
+	resp, body, err := ur.client.SendPut(url, content, &httpClientsDetails)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return errorutils.CheckError(errors.New("Distribution response: " + resp.Status + "\n" + utils.IndentJson(body)))
+	}
+
+	log.Debug("Artifactory response: ", resp.Status)
+	log.Output(utils.IndentJson(body))
+	return errorutils.CheckError(err)
+}
+
+func CreateBundleBody(createBundleParams CreateUpdateReleaseBundleParams, dryRun bool) (*ReleaseBundleBody, error) {
```

Comment:
Since this function us used by two services, maybe it should be moved into *distributionutils.go*.